### ISSUE TITLE
Web server implicitly sends 'server' header and leaks python and aiohttp versions

### ISSUE THREAD

**Issue by JanBednarik** [aio-libs/aiohttp#270]
aiohttp web server implicitly sends `server` header and there is no way to disable this behavior. Sending software name and versions in `server` header is considered as serious security issue. Server should not provide any specific info about itself in production use.

I suggest disable sending of any `server` header by default. And provide easy way to set extra headers for whole `Application` as optional `headers` argument:

```
app = web.Application(headers={'server': 'nginx'})
```

Someone may want to mask `server` as something else to improve security (mislead attackers). And easy way to set extra headers to all responses is useful feature anyway.

**Comment by fafhrd91** [comment]
I like idea of default headers, but right now it is not possible. Request handler does not have access to response headers. Response object can check if default headers are available, but I dont want to store default headers on application object. @asvetlov thoughts?

**Comment by asvetlov** [comment]
Default headers config should be not in Request but in global storage.

The headers rewriting problem may be solved by middleware without `aiohttp` modification. I'll add example in few minutes.

**Comment by asvetlov** [comment]
See 57ebeae529a408b95b1ed2775449b44c7ad20f67 for example of middleware which rewrites response headers.

**Comment by fafhrd91** [comment]
but this example works just for some use cases, it is not 100%

**Comment by JanBednarik** [comment]
Solution with middleware does not work for error responses like 404.

Maybe extra headers can be set in:

**Comment by JanBednarik** [comment]
Solution with middleware doesn't work for error responses like 404.

Extra headers can be set as kwarg of RequestHandler like:

```
requests_handler = app.make_handler(headers={'server': 'Secured Server Software'})
```

And then they can be added to response here: https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/web.py#L1537
Or maybe `response.start()` method may accept optional `headers` kwarg.

**Comment by asvetlov** [comment]
Fixed example to process web exceptions like 404: e264224

Yes, the example doesn't cover 100% use cases. It doesn't support headers overriding for started responses (streaming and websockets).

For clean solution we should store additional headers in application context (`web.Application` or `RequestHandler`), pass default headers into `web.Request` and setup those headers in `response.start(request)`. 
As an option we can store and process default headers in `HttpMessage.send_headers()` (`._add_default_header()`) but I quite don't like the second way.

**Comment by asvetlov** [comment]
Fixed by #562
